File: streamis-jobmanager/streamis-job-launcher/streamis-job-launcher-base/src/main/scala/com/webank/wedatasphere/streamis/jobmanager/launcher/job/type/JobClientType.scala
package com.webank.wedatasphere.streamis.jobmanager.launcher.job.`type`

/**
 * @author jefftlin
 * */
object JobClientType extends Enumeration {

  type ConnectType = Value
  /**
   * Attach
   */
  val ATTACH: JobClientType.Value = Value("attach")

  /**
   * Detach
   */
  val DETACH: JobClientType.Value = Value("detach")

  /**
   * Detach standalone
   */
  val DETACH_STANDALONE: JobClientType.Value = Value("detach_standalone")
}

File: streamis-jobmanager/streamis-job-launcher/streamis-job-launcher-base/src/main/java/com/webank/wedatasphere/streamis/jobmanager/launcher/enums/JobClientType.java
package com.webank.wedatasphere.streamis.jobmanager.launcher.enums;

public enum JobClientType {

    ATTACH("attach"),
    DETACH("detach"),
    DETACH_STANDALONE("detachStandalone"),

    OTHER("other");

    private String name;

    JobClientType(String name) {
        this.name = name;
    }

    public String getName() {
        return name;
    }

    public JobClientType toJobClientType(String s) {
        if ("attach".equalsIgnoreCase(s)) {
            return ATTACH;
        } else if ("detach".equalsIgnoreCase(s)) {
            return DETACH;
        } else if ("detachStandalone".equalsIgnoreCase(s)) {
            return DETACH_STANDALONE;
        } else {
            // default
            return ATTACH;
        }
    }
}
